test/pytest/test_xpedite/test_profiler/profile.py

```diff
@@ -1,180 +1,103 @@
-  if profileInfo.benchmarkPaths:
-    for i in range(0, len(profileInfo.benchmarkPaths)):
-      profileInfo.benchmarkPaths[i] = os.path.join(dataDir, profileInfo.benchmarkPaths[i])
-  return profileInfo
-
-def generateProfiles(app, profileInfo):
+def generateProfiles(app, scenario):
   """
   Start Xpedite profiling and return the profiles and result generating by profiling
-
-  @param app: An xpedite application
-  @type app: C{xpedite.app.XpediteApp} or C{xpedite.app.XpediteDormantApp}
-  @param profileInfo: Profile information generated by xpedite
-  @type profileInfo: C{xpedite.profileInfoLoader.ProfileInfo}
   """
-  from xpedite.jupyter.result   import Result
-  from xpedite.profiler         import Profiler
-  result = Result()
-  Profiler.profile(
-    app=app, profileInfo=profileInfo, reportName='XpediteTest', reportPath=None,
-    dryRun=app.dryRun, result=result, interactive=False, heartbeatInterval=1
+  from xpedite.profiler import Profiler
+  profiler = Profiler.profile(
+    app, scenario.profileInfo, REPORT_NAME, None, app.dryRun, scenario.result,
+    interactive=False, heartbeatInterval=1
   )
-  return result
+  return profiler.profiles
 
-def runXpediteReport(runId, profileInfo, sampleFilePath=None, cpuInfo=None, workspace=None):
+def runXpediteReport(runId, context, scenario, sampleFilePath=None, cpuInfoOverride=False):
   """
   Run xpedite report
-
-  @param dataFilePath: File path to override an app's sample file path
-  to a data file in the test data directory
   """
-  from xpedite.profiler.app import XpediteDormantApp
-  xpediteApp = XpediteDormantApp(profileInfo.appName, profileInfo.appHost, profileInfo.appInfo, runId, workspace)
-  if sampleFilePath:
-    xpediteApp.sampleFilePath = sampleFilePath
-  xpediteApp.start()
-  if cpuInfo:
-    xpediteApp.env.proxy.fullCpuInfo = cpuInfo
-  result = generateProfiles(xpediteApp, profileInfo)
-  xpediteApp.stop()
-  return result
+  with scenario.makeXpediteDormantApp(runId, context.workspace, sampleFilePath) as xpediteApp:
+    if cpuInfoOverride:
+      xpediteApp.env.proxy.fullCpuInfo = scenario.fullCpuInfo
+    xpediteApp.appInfoPath = os.path.join(scenario.dataDir, XPEDITE_APP_INFO_PATH)
+    profiles = generateProfiles(xpediteApp, scenario)
+    return profiles, sampleFilePath
 
-def runXpediteRecord(binary, profileInfo, txnCount, threadCount, remote=None, workspace=None):
+def runXpediteRecord(context, scenario):
   """
   Run xpedite record against a live target application process
   """
-  with TargetLauncher(binary, profileInfo, txnCount, threadCount, workspace, remote) as app:
-    result = generateProfiles(app.xpediteApp, profileInfo)
-  return app, result
+  with TargetLauncher(context, scenario) as app:
+    profiles = generateProfiles(app.xpediteApp, scenario)
+    fullCpuInfo = copy.deepcopy(app.xpediteApp.env.proxy.fullCpuInfo)
+    dataFiles = app.xpediteApp.gatherFiles(app.xpediteApp.sampleFilePattern())
+  return app, profiles, fullCpuInfo, dataFiles
```

Comment:
full CPU info is being used so it can be set in the app when generating profiles:
xpediteApp.env.proxy.fullCpuInfo = fullCpuInfo
to generate profiles with the same tsc